apps/examples/testcase/le_tc/kernel/tc_wqueue.c

```diff
@@ -0,0 +1,153 @@
+*    without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+* LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+* ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+**************************************************************************/
+
+/**************************************************************************
+* Included Files
+**************************************************************************/
+#include <tinyara/config.h>
+#include <tinyara/clock.h>
+#include <tinyara/wqueue.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include "tc_internal.h"
+
+/**************************************************************************
+* Private Definitions
+**************************************************************************/
+#if defined(CONFIG_SCHED_WORKQUEUE) || defined(CONFIG_LIB_USRWORK)
+
+/**************************************************************************
+* Private Variables
+**************************************************************************/
+
+static systime_t start_time;
+
+/**************************************************************************
+* Private Functions
+**************************************************************************/
+static void wq_test1(void *arg)
+{
+	systime_t cur_time = 0;
+	cur_time = clock_systimer();
+	printf("workqueue_test 1 : test 1 requested delay is (%u) ticks, executed delay is (%llu) ticks.\n", (uint32_t)arg, (uint64_t)cur_time - (uint64_t)start_time);
+}
+
+static void wq_test2(void *arg)
+{
+	systime_t cur_time = 0;
+	cur_time = clock_systimer();
+	printf("workqueue_test 2 : test 2 requested delay is (%u) ticks, executed delay is (%llu) ticks.\n", (uint32_t)arg, (uint64_t)cur_time - (uint64_t)start_time);
+}
+
+static void wq_test3(void *arg)
+{
+	systime_t cur_time = 0;
+	cur_time = clock_systimer();
+	printf("workqueue_test 3 : test 3 requested delay is (%u) ticks, executed delay is (%llu) ticks.\n", (uint32_t)arg, (uint64_t)cur_time - (uint64_t)start_time);
+}
+/**************************************************************************
+* Public Functions
+**************************************************************************/
+#if defined(CONFIG_SCHED_HPWORK) || defined(CONFIG_SCHED_LPWORK)
+static void tc_wqueue_cancel(void)
```

Comment:
Why is function name (tc name) cancel? only cancel is main content? Only cancel without any normal operation of wqueue doesn't make sense.

Reply:
Okay , Can I name it as tc_wqueue_create_cancel ??

Reply:
Our guidance is adding function names so that ```tc_wqueue_work_queue_cancel``` is correct.

Reply:
Okay